app/assets/javascripts/admin/addon/routes/admin-badges/show.js

```diff
@@ -27,51 +23,14 @@ export default class AdminBadgesShowRoute extends Route {
     );
   }
 
-  setupController(controller, model) {
+  setupController(controller) {
     super.setupController(...arguments);
-    if (model.image_url) {
-      controller.showImageUploader();
-    } else if (model.icon) {
-      controller.showIconSelector();
-    }
-  }
 
-  @action
-  editGroupings() {
-    const model = this.controllerFor("admin-badges").get("badgeGroupings");
-    this.modal.show(EditBadgeGroupingsModal, {
-      model: {
-        badgeGroupings: model,
-        updateGroupings: this.updateGroupings,
-      },
-    });
+    controller.setup();
   }
 
   @action
   updateGroupings(groupings) {
     this.controllerFor("admin-badges").set("badgeGroupings", groupings);
   }
-
-  @action
-  async preview(badge, explain) {
```

Comment:
This has been moved into the controller, to avoid the need for `badge.set("preview_loading", ...)` which is not possible with FormKit as we don't manipulate the model object and also because it's not a good pattern.